engine/src/flutter/shell/platform/embedder/embedder.h

```diff
@@ -258,6 +262,94 @@ typedef enum {
   kFlutterSemanticsFlagIsRequired = 1 << 30,
 } FlutterSemanticsFlag;
 
+typedef struct {
+  /// The size of this struct. Must be sizeof(FlutterSemanticsFlags).
+  size_t struct_size;
+  /// The semantics node has the quality of either being "checked" or
+  /// "unchecked".
+  bool has_checked_state;
+  /// Whether a semantics node is checked.
+  bool is_checked;
```

Comment:
Do we want to do something nicer here? We could have a single member for checked state if we introduced an enum with values "none", "true", "false". I don't feel strongly about this at all, feel free to skip.

Reply:
but i think it can be in another PR to update embedder and engine and frameworrk all together.